src/game/AI/ScriptDevAI/scripts/northrend/ulduar/ulduar/boss_auriaya.cpp

```diff
@@ -63,329 +62,233 @@ enum
-            if (m_uiReviveDelayTimer <= uiDiff)
-            {
-                // ToDo: figure out how to enable the ressurrect animation
-                DoCastSpellIfCan(m_creature, SPELL_FULL_HEAL, CAST_TRIGGERED);
-                m_uiReviveDelayTimer = 0;
-                m_uiAttackDelayTimer = 3000;
-            }
-            else
-                m_uiReviveDelayTimer -= uiDiff;
-
-            // No Further action while faking
-            return;
-        }
-
-        if (m_uiAttackDelayTimer)
+        switch (action)
         {
-            if (m_uiAttackDelayTimer <= uiDiff)
+            case FERAL_DEFENDER_FERAL_RUSH:
             {
-                DoResetThreat();
-                m_creature->RemoveAurasDueToSpell(SPELL_FEIGN_DEATH);
-                m_creature->SetStandState(UNIT_STAND_STATE_STAND);
-                m_creature->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_UNINTERACTIBLE);
-
-                m_uiPounceTimer = urand(9000, 10000);
-                m_uiFeralRushCount = 0;
-                m_uiFeralRushTimer = 1000;
-                m_uiAttackDelayTimer = 0;
-            }
-            else
-                m_uiAttackDelayTimer -= uiDiff;
-
-            // No Further action while faking
-            return;
-        }
-
-        if (m_uiPounceTimer < uiDiff)
-        {
-            if (Unit* pTarget = m_creature->SelectAttackingTarget(ATTACKING_TARGET_RANDOM, 0))
-            {
-                if (DoCastSpellIfCan(pTarget, m_bIsRegularMode ? SPELL_FERAL_POUNCE : SPELL_FERAL_POUNCE_H) == CAST_OK)
-                    m_uiPounceTimer = urand(13000, 16000);
-            }
-        }
-        else
-            m_uiPounceTimer -= uiDiff;
-
-        if (m_uiFeralRushTimer < uiDiff)
-        {
-            if (DoCastSpellIfCan(m_creature, m_bIsRegularMode ? SPELL_FERAL_RUSH : SPELL_FERAL_RUSH_H) == CAST_OK)
-            {
-                ++m_uiFeralRushCount;
-
-                if (m_uiFeralRushCount == m_uiMaxFeralRush)
+                if (DoCastSpellIfCan(nullptr, m_isRegularMode ? SPELL_FERAL_RUSH : SPELL_FERAL_RUSH_H) == CAST_OK)
                 {
-                    m_uiFeralRushCount = 0;
-                    m_uiFeralRushTimer = 12000;
+                    ++m_feralRushCount;
+                    if (m_feralRushCount < m_maxFeralRush)
+                        ResetCombatAction(action, 400ms);
+                    else
+                    {
+                        m_feralRushCount = 0;
+                        ResetCombatAction(action, 12s);
+                    }
                 }
-                else
-                    m_uiFeralRushTimer = 400;
+                return;
             }
         }
-        else
-            m_uiFeralRushTimer -= uiDiff;
+    };
+};
 
-        DoMeleeAttackIfReady();
+// 64386 - Terrifying Screech
+struct TerrifyingScreech : public SpellScript
+{
+    void OnSuccessfulFinish(Spell* spell) const override
+    {
+        Creature* caster = dynamic_cast<Creature*>(spell->GetCaster());
```

Comment:
You do not need a creature cast here. AI is already in unit and IsCreature check is enough instead.

Reply:
True, but I found it visually more simple to have it as one line than to do
```cpp
Creature* caster;
if (spell->GetCaster()->IsCreature())
    caster = static_cast<Creature*>(spell->GetCaster());
```

Reply:
ehh I guess
```cpp
    void OnSuccessfulFinish(Spell* spell) const override
    {
        Unit* caster = spell->GetCaster();
        if (!caster || !caster->AI() || !caster->IsCreature())
            return;
        caster->AI()->DoCastSpellIfCan(nullptr, caster->GetMap()->IsRegularDifficulty() ? SPELL_SENTINEL_BLAST : SPELL_SENTINEL_BLAST_H);
    }
```
works too